tools/rdm/TestDefinitions.py

```diff
@@ -6473,14 +6433,177 @@ class GetListInterfacesWithData(TestMixins.GetWithDataMixin,
   CATEGORY = TestCategory.ERROR_CONDITIONS
   PID = 'LIST_INTERFACES'
 
-class SetListInterfaces(ResponderTestFixture):
+class SetListInterfaces(TestMixins.UnsupportedSetMixin,
+                        OptionalParameterTestFixture):
   """Attempt to SET list interfaces."""
   CATEGORY = TestCategory.ERROR_CONDITIONS
   PID = 'LIST_INTERFACES'
 
+# DNS_HOSTNAME
+#------------------------------------------------------------------------------
+class GetDNSHostname(TestMixins.GetStringMixin,
+                     OptionalParameterTestFixture):
+  """GET the DNS hostname."""
+  CATEGORY = TestCategory.IP_DNS_CONFIGURATION
+  PID = 'DNS_HOSTNAME'
+  EXPECTED_FIELD = 'dns_hostname'
+  ALLOWED_NACK = RDMNack.NR_HARDWARE_FAULT
+  MIN_LENGTH = RDM_MIN_HOSTNAME_LENGTH
+  MAX_LENGTH = RDM_MAX_HOSTNAME_LENGTH
+
+class GetDNSHostnameWithData(TestMixins.GetWithDataMixin,
+                             OptionalParameterTestFixture):
+  """Get DNS hostname with param data."""
+  CATEGORY = TestCategory.ERROR_CONDITIONS
+  PID = 'DNS_HOSTNAME'
+  # Allow NR_HARDWARE_FAULT in case they're checking length then PDL
+  ALLOWED_NACK = RDMNack.NR_HARDWARE_FAULT
```

Comment:
It seems we do for starters, as we run a custom check regarding the length, then pass it onto our general string handler. Shall we therefore leave it in, given others may code it in the same way? Or is the updated E1.20 going to mandate checking for format errors first?